src/explore-education-statistics-admin/src/components/RouteLeavingGuard.tsx

```diff
@@ -0,0 +1,72 @@
+import ModalConfirm from '@common/components/ModalConfirm';
+import useToggle from '@common/hooks/useToggle';
+import React, { useEffect, useState } from 'react';
+import { Prompt, useLocation } from 'react-router';
+import { useHistory } from 'react-router-dom';
+
+interface Props {
+  blockRouteChange: boolean;
+}
+
+const RouteLeavingGuard = ({ blockRouteChange = false }: Props) => {
+  const location = useLocation();
+  const history = useHistory();
+  const [lastLocation, setLastLocation] = useState(location);
+  const [showModal, toggleShowModal] = useToggle(false);
+  const [confirmedNavigation, toggleConfirmedNavigation] = useToggle(false);
+
+  // Block non-react routes
+  useEffect(() => {
+    const handleBeforeUnload = (event: BeforeUnloadEvent) => {
+      if (blockRouteChange) {
+        event.preventDefault();
+        // eslint-disable-next-line no-param-reassign
+        event.returnValue = '';
+      }
+    };
+
+    window.addEventListener('beforeunload', handleBeforeUnload);
+
+    return () => window.removeEventListener('beforeunload', handleBeforeUnload);
+  }, [blockRouteChange]);
+
+  // Block react routes
+  useEffect(() => {
+    if (confirmedNavigation && lastLocation) {
+      toggleConfirmedNavigation.off();
+      history.push(lastLocation.pathname);
+    }
+  }, [confirmedNavigation, lastLocation, history, toggleConfirmedNavigation]);
+
+  return (
+    <>
+      <Prompt
+        when={blockRouteChange}
+        message={nextLocation => {
+          if (!confirmedNavigation && blockRouteChange) {
+            setLastLocation(nextLocation);
+            toggleShowModal.on();
+            return false;
+          }
+          return true;
+        }}
+      />
+      <ModalConfirm
+        title="There are unsaved changes"
```

Comment:
We probably want to make this configurable as a prop

Reply:
Done